tests/net/lib/mqtt_subscriber/net.mqtt.subscriber fails to build on cc3220sf_launchxl

Running:

`./scripts/sanitycheck -p cc3220sf_launchxl -s tests/net/lib/mqtt_subscriber/net.mqtt.subscriber`

I get:

```
[ 89%] Building C object modules/ti/simplelink/CMakeFiles/..__modules__hal__ti__simplelink.dir/kernel/zephyr/dpl/dpl.c.obj
In file included from /home/galak/git/zephyr/include/device.h:22,
                 from /home/galak/git/modules/hal/ti/simplelink/kernel/zephyr/dpl/dpl.c:12:
/home/galak/git/zephyr/include/init.h:89:11: error: initialization of 'int (*)(const struct device *)' from incompatible pointer type 'int (*)(struct device *)' [-Werror=incompatible-pointer-types]
   89 |   .init = (_init_fn),     \
      |           ^
/home/galak/git/zephyr/include/init.h:136:2: note: in expansion of macro 'Z_INIT_ENTRY_DEFINE'
  136 |  Z_INIT_ENTRY_DEFINE(Z_SYS_NAME(_init_fn), _init_fn, NULL, _level, _prio)
      |  ^~~~~~~~~~~~~~~~~~~
/home/galak/git/modules/hal/ti/simplelink/kernel/zephyr/dpl/dpl.c:63:1: note: in expansion of macro 'SYS_INIT'
   63 | SYS_INIT(dpl_zephyr_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
      | ^~~~~~~~
/home/galak/git/zephyr/include/init.h:89:11: note: (near initialization for '__init_sys_init_dpl_zephyr_init2.init')
   89 |   .init = (_init_fn),     \
      |           ^
/home/galak/git/zephyr/include/init.h:136:2: note: in expansion of macro 'Z_INIT_ENTRY_DEFINE'
  136 |  Z_INIT_ENTRY_DEFINE(Z_SYS_NAME(_init_fn), _init_fn, NULL, _level, _prio)
      |  ^~~~~~~~~~~~~~~~~~~
/home/galak/git/modules/hal/ti/simplelink/kernel/zephyr/dpl/dpl.c:63:1: note: in expansion of macro 'SYS_INIT'
   63 | SYS_INIT(dpl_zephyr_init, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
      | ^~~~~~~~
cc1: all warnings being treated as errors
````

--- a/west.yml
+++ b/west.yml
@@ -77,7 +77,7 @@ manifest:
       revision: cea57f86d3ade46c0b99dab32b2ac63435ed0693
       path: modules/hal/stm32
     - name: hal_ti
-      revision: deb246ade0d4d64f470482017caec600cc48086e
+      revision: 405dfc8faba13e01c1cb9e29e70b31b50f71d117
       path: modules/hal/ti
     - name: libmetal
       revision: 87e9e7f2c5b4e238236fe703db61ba23e48dc2ef

